compiler/src/dotty/tools/dotc/transform/init/Objects.scala

```diff
@@ -625,7 +653,7 @@ object Objects:
             // eval(ddef.rhs, ref, cls, cacheResult = true)
             cache.cachedEval(ref, ddef.rhs, cacheResult = true) { expr =>
               Returns.installHandler(meth)
-              val res = cases(expr, thisV, cls)
+              val res = cases(expr, thisV.asInstanceOf[ValueElement], cls)
```

Comment:
Looks like `thisV` here comes from `Env.resolveEnv`. Can we guarantee that `Env.resolveEnv` never returns a `RefSet`?